#169817862:user signup

mujohn26 opened the pull request:
#### What does this PR do?

-  user should be able to create account

#### Description of Task to be completed?

-  user must type the fields to create
-  user will receive JWT after creating account

#### How should this be manually tested?

-  clone it 
- run npm i to install dependencies
- run `npm  run start`

#### Any background context you want to provide?

- all fields are required you can not create account with invalid input

#### What are the relevant pivotal tracker stories?

[#169817862](https://www.pivotaltracker.com/story/show/169817862)

#### Screenshots (if appropriate)
<img width="1150" alt="Screen Shot 2019-12-19 at 09 18 49" src="https://user-images.githubusercontent.com/52497006/71165538-b0b05000-2259-11ea-863b-a8df5a3d4df6.png">



#### Questions:

N/A

shemaeric1234 commented:
Remove coverage files

aanchirinah commented on coverage/base.css:
```diff
@@ -0,0 +1,224 @@
+body, html {
```
### Nitpick

- We don't need this commited to Github

mujohn26 replied:
I resolved it

aanchirinah commented on coverage/block-navigation.js:
```diff
@@ -0,0 +1,79 @@
+/* eslint-disable */
```
### Critical

- Include the coverage directory inside `.gitignore`

mujohn26 replied:
yeah I have done

aanchirinah reviewed (changes requested):
Use a NodeJs `.gitignore` template

jo-rdan reviewed (approved):
You did good John!

aanchirinah commented on src/controllers/userController.js:
```diff
@@ -0,0 +1,66 @@
+   * @param {Object} res The response object
+   * @returns {Object} A user object with selected fields
+   */
+  static async signup(req, res) {
```
### Nitpick

- You are using `Async` but I see no `Await block of code`.

mujohn26 replied:
I solved it

aanchirinah commented on src/helpers/responseHandler.js:
```diff
@@ -0,0 +1,38 @@
+ */
+class response {
+  /**
+   * getting response
```
### Nitpick

- Not descriptive enough

mujohn26 replied:
I resolved it by adding more descriptive

aanchirinah commented on src/helpers/responseHandler.js:
```diff
@@ -0,0 +1,38 @@
+  }
+
+  /**
+   * getting response
```
### Nitpick

- Not descriptive enough

aanchirinah commented on src/services/UserService.js:
```diff
@@ -0,0 +1,17 @@
+import { user } from '../database/models';
+import Queries from './Queries';
+/**
+* class for responses
```
### Nitpick 

- Don't think that's a good description

mujohn26 replied:
I added a lot more description

aanchirinah commented on src/tests/user/userData.js:
```diff
@@ -0,0 +1,13 @@
+const userData = {
```
### Nitpick

- You can also tryout `faker.js`

mujohn26 replied:
I implemented faker

aanchirinah reviewed (changes requested):
Good job , please attend to requested changes.

shemaeric1234 reviewed (commented):
Validation is not working properly and coverage decreased

mujohn26 committed:
ft(signup): signup account

user should create account
swagger documentation
solving conflicts
[continue #169817862]

dcDalin commented on src/controllers/user.controller.js:
```diff
@@ -1,11 +1,133 @@
+        token,
+      };
+      const userName = firstName;
+      const view = `<!DOCTYPE html>
```
I think it would be nice to set create email templates in another folder.

mujohn26 replied:
Thanks,I solved it !
